azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/ChildListFlattener.java

```diff
@@ -0,0 +1,181 @@
+package com.microsoft.azure.management.compute.implementation;
+
+import com.microsoft.azure.CloudException;
+import com.microsoft.azure.Page;
+import com.microsoft.azure.PagedList;
+import com.microsoft.rest.RestException;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * {@link ChildListFlattener} that can take a paged list of parents and flatten their child lists
+ * as a single lazy paged list.
+ *
+ * @param <T> the type of parent paged list item
+ * @param <U> the type of child paged list item
+ */
+public final class ChildListFlattener<T, U> {
```

Comment:
right, I was thinking that or suggest package-internal :-) Since this there is only one specific case we're dealing with so far, I figured it's better to keep the exposure to a minimum. If we come across another need elsewhere, we can make public and move to resources. At least making it internal now will prevent customers from taking a premature dependency on it.